experimental/dds/tree2/src/codec/typeboxValidator.ts

```diff
@@ -0,0 +1,28 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// This export is documented as supported in typebox's documentation.
+// eslint-disable-next-line import/no-internal-modules
+import { TypeCompiler } from "@sinclair/typebox/compiler";
+import type { Static, TSchema } from "@sinclair/typebox";
+import { JsonValidator } from "./codec";
+
+/**
+ * A {@link JsonValidator} implementation which uses TypeBox's JSON schema validator.
+ * @alpha
+ * @privateRemarks - Take care to not reference this validator directly in SharedTree code:
```

Comment:
This makes me think typeboxValidator should be moved out of codec (which lots of things can and should depend on) into somewhere where it is an error if its referenced by most stuff.

A bare file in feature-libraries makes a lot of logical sense and is less legal to depend on than here.

Another option is to make a new place for it, fenced such that nothing can import it except the root index. (ex: add a "external-utilities" folder or something like that. Another option is to just throw in in shared-tree since nothing other than shared tree itself and index can depend on that (nice defensively, but logically kinda odd).